x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/function/FunctionRegistry.java

```diff
@@ -118,18 +118,21 @@ public boolean functionExists(String functionName) {
         // It is worth double checking if we need this copy. These are immutable anyway.
         Pattern p = Strings.hasText(pattern) ? Pattern.compile(normalize(pattern)) : null;
         return defs.entrySet().stream()
-                .filter(e -> p == null || p.matcher(e.getKey()).matches())
-                .map(e -> new FunctionDefinition(e.getKey(), emptyList(),
-                        e.getValue().clazz(), e.getValue().extractViable(), e.getValue().builder()))
-                .collect(toList());
+            .filter(e -> p == null || p.matcher(e.getKey()).matches())
+            .map(e -> cloneDefinition(e.getKey(), e.getValue()))
+            .collect(toList());
+    }
+
+    protected FunctionDefinition cloneDefinition(String name, FunctionDefinition definition) {
+        return new FunctionDefinition(name, emptyList(), definition.clazz(), definition.extractViable(), definition.builder());
     }
 
     /**
      * Build a {@linkplain FunctionDefinition} for a no-argument function that
      * is not aware of time zone and does not support {@code DISTINCT}.
      */
-    protected static <T extends Function> FunctionDefinition def(Class<T> function,
-            java.util.function.Function<Source, T> ctorRef, String... names) {
+    public static <T extends Function> FunctionDefinition def(Class<T> function,
+                                                              java.util.function.Function<Source, T> ctorRef, String... names) {
```

Comment:
This class needs to be refactored in a follow-up. The methods were made public from protected to make them a bit consistent - some methods were public but the interface in the arguments was protected while methods without interfaces were protected...